pkg/provider/provider.go

```diff
@@ -307,7 +316,11 @@ func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib ma
 		}
 
 		// these files will be returned to the CSI driver as part of gRPC response
-		files[fileName] = objectContent
+		files = append(files, SecretFile{
+			Path:     fileName,
+			Content:  objectContent,
+			FileMode: keyVaultObject.Permission,
```

Comment:
We can set the default file permission we get from the driver if it's not overridden by the user with `filePermission`